lib/src/auth/frappe/frappe.auth.controller.dart

```diff
@@ -691,14 +691,59 @@ class FrappeAuthController extends AuthController<FrappeSessionStatusInfo> {
     String state,
   }) async {
     await getFrappe().checkAppInstalled(features: ['loginViaGoogle']);
+
+    assert(code != null && code.isNotEmpty, 'Code cannot be empty');
+
+    final response = await Request.initiateRequest(
+        url: config.hostUrl,
+        method: HttpMethod.POST,
+        contentType: ContentTypeLiterals.APPLICATION_JSON,
+        data: <String, dynamic>{
+          'cmd': 'renovation_core.oauth.login_via_google',
+          'code': code,
+          'state': state,
+          'use_jwt': _useJwt
+        },
+        isFrappeResponse: false);
+
+    FrappeSessionStatusInfo sessionStatusInfo;
+    if (response.isSuccess) {
+      sessionStatusInfo = FrappeSessionStatusInfo.fromJson(
+          Request.convertToMap(response.rawResponse));
+      sessionStatusInfo.rawSession = Request.convertToMap(response.rawResponse);
+    }
+    updateSession(
+        sessionStatus: sessionStatusInfo, loggedIn: response.isSuccess);
+    return response.isSuccess
+        ? RequestResponse.success(sessionStatusInfo,
+            rawResponse: response.rawResponse)
+        : RequestResponse.fail(response.error);
+  }
+
+  /// Logins in using Apple Auth code.
+  ///
+  /// In addition to the code, the option must be specified [APPLE_OPTION].
+  ///
+  /// Optionally pass the [state] which is usually a JWT or base64 encoded data.
+  @override
+  Future<RequestResponse<FrappeSessionStatusInfo>> loginViaApple({
+    @required String code,
+    @required APPLE_OPTION option,
+    String state,
+  }) async {
+    await getFrappe().checkAppInstalled(features: ['loginViaApple']);
+
     assert(code != null && code.isNotEmpty, 'Code cannot be empty');
+    assert(option != null, 'Apple option must be specified');
+
     final response = await Request.initiateRequest(
         url: config.hostUrl,
         method: HttpMethod.POST,
         contentType: ContentTypeLiterals.APPLICATION_JSON,
         data: <String, dynamic>{
           'cmd': 'renovation_core.oauth.login_via_google',
```

Comment:
cmd meant for Apple, has google's cmd.

Reply:
Good catch. This is the disadvantage of copy-pasting 😛